datashader/tests/test_raster.py

```diff
@@ -138,12 +138,12 @@ def test_raster_both_ascending_partial_range():
     ys = np.arange(5)
     arr = xs*ys[np.newaxis].T
     xarr = xr.DataArray(arr, coords={'X': xs, 'Y': ys}, dims=['Y', 'X'])
-    cvs = ds.Canvas(9, 4, x_range=(-.5, 8.5), y_range=(-.5, 3.5))
+    cvs = ds.Canvas(7, 3, x_range=(.5, 7.5), y_range=(.5, 3.5))
```

Comment:
Can you motivate this change to the tests?  Is it now testing something more than it was before?

Reply:
Yes, it's now slicing different numbers of elements along each edge. Previously I was only slicing off the upper bound, which let some bugs slip past.

Reply:
Sounds good.